### ISSUE TITLE
Crafting UI improvements

### ISSUE THREAD

**Issue by johnkslg** [StrangeLoopGames/EcoIssues#14283]
![image](https://user-images.githubusercontent.com/3536496/71597687-0fe34f80-2af9-11ea-99fc-ea40558bab72.png)
- [ ] This is getting a bit crowded and sloppy looking. What we need is a system that sorts these, so that they all appear pointing at the same spot, but stacked vertically. Check WorldMarkerManager.UpdateMarkerList, this sorts world markers. We should make a generic type 'OverlayText' that world markers and the popup instructions both interit from, and make that sort properly, probably in a new OverlayTextManager.  Create a design for this and get signoff from me and Sergey.

![image](https://user-images.githubusercontent.com/3536496/71597790-681a5180-2af9-11ea-994d-ce1d16f6c7e2.png)
- [ ] When adding labor, this popped up correctly, however I'd like to change a few things.  
- Make the camera force-look at the icon to ensure that it's seen by the player (share the code used when you click on the workorders at the top of the screen).
- The tween looks like BouncInOut, use something tighter like QuadOut.
- When adding labor, I saw text '+ X labor', but it started before the zoom, and then faded too quick, make it stay visible for longer.  The 'complete' text also had this issue.
- Adding the labor made the bar go up immediately, instead of waiting till it had zoomed in.
- Adding ingredients increased the bars, but it should also say '+ <ingredients> added'
- Right clicking when there was no labor to add still zoomed

![image](https://user-images.githubusercontent.com/3536496/71597960-03abc200-2afa-11ea-8bd2-2ecf6c2f482d.png)
- [ ] Changed my thoughts on this, its confusing what is added, needed, and total. I think we can simplify this further, and just say what's still needed, and not list requirements already satisfied at all.  Change it to be this way.

![image](https://user-images.githubusercontent.com/3536496/71598121-83d22780-2afa-11ea-8233-f0f125ca5f7c.png)
- [x] Lets make this display only needed as well. Make it say 'X calories of labor needed'

![image](https://user-images.githubusercontent.com/3536496/71598063-4cfc1180-2afa-11ea-8819-93fb60a69531.png)
- [ ] Add a tooltip on the 'work' button when its greyed out explaining why you can't add work.  Something like 'You can't add work until more ingredients are available' or 'All needed labor has been performed on this work order.'

![image](https://user-images.githubusercontent.com/3536496/71598620-6736ef00-2afc-11ea-97bd-edd18771b7e9.png)
- [ ] Overlay is only showing the finished order, even though a new one is started. Ideally it shows both, or it can just show the active unfinished one.  
- [ ] Right clicking with a handful of wood wouldn't add to the unfinished workorder on the table, when there is a finished one sitting there. 

![image](https://user-images.githubusercontent.com/3536496/71598688-b715b600-2afc-11ea-9951-49e546923363.png)
- [ ] Work button was enabled despite the work order being finished, and collection button is missing.

![image](https://user-images.githubusercontent.com/3536496/71598701-cac11c80-2afc-11ea-8ea0-a08835ab7074.png)
 - [ ] I attempted to cancel the finished order, nothing happened.   Cancel should not be available if an order is finished at all. 

![image](https://user-images.githubusercontent.com/3536496/71600895-dfee7900-2b05-11ea-8829-c619e0050f0b.png)
- [ ] Bars not displaying correctly, I see green and yellow in the table ui, but only one in the popup ui

Basically I want to make sure the interface feels really inuitive and responsive, and helps communicate the complexities of the system to a user who doesn't know how it works yet.  Optimize for this, putting in polish to improve this goal beyond what I've listed here

**Comment by SlayksWood** [comment]
Missing icons in last builds.
![image](https://user-images.githubusercontent.com/45708377/72721598-7c4bee80-3b8d-11ea-9e15-c298eef7d3dd.png)

**Comment by johnkslg** [comment]
Refactored all of this, so closing this out.  For new feedback, please start a new issue.
